poseidon/controllers/faucet/parser.py

```diff
@@ -86,57 +86,45 @@ def parse_rules(config_file):
         obj_doc = Parser().yaml_in(config_file)
         return obj_doc
 
-    def config(self, config_file, action, port, switch, rules_file=None, endpoints=None, force_apply_rules=None, force_remove_rules=None, coprocess_rules_files=None):
+    def check_mirror(self, config_file, switch, port, obj_doc):
+        if not obj_doc:
+            return False
+        if not self.mirror_ports:
+            self.logger.error('Unable to mirror, no mirror ports defined')
+            return False
+        if 'dps' not in obj_doc:
+            self.logger.warning(f'Unable to find switches in {config_file}')
+            return False
+        else:
+            for s in obj_doc['dps']:
+                if (switch == s and s in self.mirror_ports and
+                    'interfaces' in obj_doc['dps'][s] and
+                    port in obj_doc['dps'][s]['interfaces'] and
+                        self.mirror_ports[s] in obj_doc['dps'][s]['interfaces']):
+                    return s
+        self.logger.warning(f'No switch/port match found to mirror from in '
+                            'the configs or mirror port not defined on that '
+                            'switch: {switch} {obj_doc}')
+
+    def config(self, config_file, action, port, switch, rules_file=None,
+               endpoints=None, force_apply_rules=None, force_remove_rules=None,
+               coprocess_rules_files=None):
         status = [True, []]
         switch_found = None
         config_file = Parser().get_config_file(config_file)
         obj_doc = Parser().yaml_in(config_file)
 
-        if not obj_doc:
-            return False
+        switch_found = self.check_mirror(config_file, switch, port, obj_doc)
 
         if action == 'mirror' or action == 'unmirror':
-            ok = True
-            if not self.mirror_ports:
-                self.logger.error('Unable to mirror, no mirror ports defined')
-                return False
-            if 'dps' not in obj_doc:
-                self.logger.warning(
-                    'Unable to find switch configs in {0}'.format(config_file))
-                ok = False
-            else:
-                for s in obj_doc['dps']:
-                    if switch == s:
-                        switch_found = s
-            if not switch_found:
-                self.logger.warning('No switch match found to mirror '
-                                    'from in the configs. switch: {0} {1}'.format(switch, str(obj_doc)))
-                ok = False
-            else:
-                if switch_found not in self.mirror_ports:
-                    self.logger.warning('Unable to mirror {0} on {1}, mirror port not defined on that switch'.format(
-                        str(port), str(switch_found)))
-                    ok = False
+            if switch_found:
+                if 'mirror' in obj_doc['dps'][switch_found]['interfaces'][self.mirror_ports[switch_found]]:
```

Comment:
Maybe take a reference to obj_doc['dps'][switch_found]['interfaces']? Is used quite a bit.